### PR TITLE
Dev/189 [UI] New Problem List

### PR DESCRIPTION
**Pull request by Wally5077**
resolve: #189 

**Visible Problems**

---

**Invisible Problems**

---

**Archive Problems**

### CODE REVIEW

File: src/app/problem/ProblemList.js [hunk 1]
```diff
@@ -34,18 +35,67 @@ const ProblemList = () => {
...
+        return (
+            <div className="is-flex is-justify-content-center"
+                 style={{position: "absolute", top: "80px", left: "780px"}}>
+                <div className="select" id="filter" style={{marginLeft: "40px", marginRight: "8px"}}>
+                    <select style={{border: "none", color: "#676877"}}>
+                        {filterItems?.map(name => <option key={name}>{name}</option>)}
+                    </select>
+                </div>
+                <input className="search" placeholder="&#xF002;" type="text" id="searchBar"/>
+                <CreateButton onClick={() => setShowCreateProblemModal(true)}/>
```
Comment (Chaoyuuu) [inline src/app/problem/ProblemList.js:48]: Where did you use `id="searchBar"`?
Remove it if you didn't use it. 👍🏻

File: src/app/problem/ProblemList.js [hunk 2]
```diff
@@ -34,18 +35,67 @@ const ProblemList = () => {
         history.push(`/problems/${problemId}/edit`);
     };
 
-    if (!problems)
+    const ProblemFilterSearchBar = ({filterItems}) => {
+        return (
+            <div className="is-flex is-justify-content-center"
+                 style={{position: "absolute", top: "80px", left: "780px"}}>
+                <div className="select" id="filter" style={{marginLeft: "40px", marginRight: "8px"}}>
+                    <select style={{border: "none", color: "#676877"}}>
```
Comment (Chaoyuuu) [inline src/app/problem/ProblemList.js:43]: Where did you use `id="filter"`?
Remove it if you didn't use it. 👍🏻

Comment (Chaoyuuu) [inline src/app/problem/ProblemList.js:43]: It will be better to move these CSS styles to the `problemList.scss`.

**Review by Chaoyuuu — CHANGES_REQUESTED** [review]
Review done!

File: src/app/problem/ProblemList.js [hunk 3]
```diff
@@ -34,34 +35,86 @@ const ProblemList = () => {
...
 
-    if (!problems)
+    const ProblemFilterSearchBar = ({filterItems}) => {
+        return (
+            <div className="is-flex is-justify-content-center filterSearchBar">
+                <div className="select">
+                    <select>
+                        {filterItems?.map(name => <option key={name}>{name}</option>)}
+                    </select>
+                </div>
```
Comment (Chaoyuuu) [inline src/app/problem/ProblemList.js:45]: I guess we don't need `<div>` here. 🤔 
Remove it if we don't need it.

**Review by Chaoyuuu — CHANGES_REQUESTED** [review]
Last comment! Thanks!

**Review by Johnny850807 — APPROVED** [review]
lGTM

File: src/app/problem/ProblemList.js (hunk 3 above)
Comment (Wally5077) [reply src/app/problem/ProblemList.js:46]: It has its own style in ProblemList.scss flie.

**Review by Chaoyuuu — APPROVED** [review]
LGTM

**Pull request merged**